internal/blocktx/background_workers.go

```diff
@@ -0,0 +1,100 @@
+package blocktx
+
+import (
+	"context"
+	"log/slog"
+	"sync"
+	"time"
+
+	"github.com/bitcoin-sv/arc/internal/blocktx/store"
+	"github.com/libsv/go-p2p"
+)
+
+type BackgroundWorkers struct {
+	l *slog.Logger
+	s store.BlocktxStore
+
+	workersWg sync.WaitGroup
+	ctx       context.Context
+	cancelAll func()
+}
+
+func NewBackgroundWorkers(store store.BlocktxStore, logger *slog.Logger) *BackgroundWorkers {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	return &BackgroundWorkers{
+		s: store,
+		l: logger.With(slog.String("module", "background workers")),
+
+		ctx:       ctx,
+		cancelAll: cancel,
+	}
+}
+
+func (w *BackgroundWorkers) GracefulStop() {
+	w.l.Info("Shutting down")
+
+	w.cancelAll()
+	w.workersWg.Wait()
+
+	w.l.Info("Shutted down")
+}
+
+func (w *BackgroundWorkers) StartFillGaps(peers []p2p.PeerI, interval time.Duration, retentionDays int, blockRequestCh chan<- BlockRequest) {
+	w.workersWg.Add(1)
+
+	go func() {
+		defer w.workersWg.Done()
+
+		t := time.NewTicker(interval)
+		i := 0
+
+		for {
+			select {
+			case <-t.C:
+				err := w.fillGaps(peers[i%len(peers)], retentionDays, blockRequestCh)
```

Comment:
Maybe this would be better so we don't overflow the `i`? 
```suggestion
				i = i%len(peers)
				err := w.fillGaps(peers[i], retentionDays, blockRequestCh)
```